packages/composables/src/hooks/useUser.ts

```diff
@@ -60,6 +80,35 @@ export const useUser = (): UseUser => {
     }
   };
 
+  const register = async (params: RegisterCustomerParams): Promise<boolean> => {
+    loading.value = true;
+    error.value = null;
+    try {
+      await apiRegister({
```

Comment:
is RegisterCustomerParams different from apiRegister? couldn't it just be:
```suggestion
      await apiRegister(params);
```
?

Reply:
There are different, I change some structure of that interface to fill the billing address also
```js
        {
        firstName: params.firstName,
        lastName: params.lastName,
       salutationId: params.salutationId,
       // ...
        billingAddress: {
          firstName: params.firstName,
          salutationId: params.salutationId,
          lastName: params.lastName,
         // ..
        }
```
Do you have any other idea to prevent from code redundancy?

Reply:
you should be consistent with API structure, otherwise, when someone would want to add different addresses (billing, shipping) during registration your structure wouldn't allow that.
It all depends on which interface we want to give users for registration. If is different than API clients, then we need to map them.

Reply:
I assumed that while the registration user provide only minimum informations that are required, and later while the ordering they will fill the rest needed data. In this case I change the interface structure to simplify it and map the rest informations.

Reply:
Ok, I will move that logic to the fronted component